### PR TITLE
Add API test cases and test data

### PR DESCRIPTION
**Pull request by ikoshel**
New test cases have been added to `apiTests.spec.js` to facilitate API testing with Cypress. Ensuring the correct API functions such as GET, POST, PUT, DELETE are working as expected with specific conditions such as non-existent IDs, incorrect data etc. Test data for these test cases has been added to `apiTestData.json`, providing easily maintainable and structured test data.

### Please read and mark the following check list before creating a pull request (check one with "x"):

 - [ ] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/ngx-admin/blob/master/CONTRIBUTING.md) guide.
 - [ ] I read and followed the [New Feature Checklist](https://github.com/akveo/ngx-admin/blob/master/DEV_DOCS.md#new-feature-checklist) guide.
 
 #### Short description of what this resolves:

### CODE REVIEW

File: cypress/e2e/api/apiTests.spec.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+const API_ENDPOINT = 'https://jsonplaceholder.typicode.com/posts';
```
Comment (sttaran) [inline cypress/e2e/api/apiTests.spec.js:1]: I would set baseUrl in a config but it's okay for now

File: cypress/e2e/api/apiTests.spec.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+  it('Get post by ID', () => {
+    cy.request({
+      method: 'GET',
+      url: `${API_ENDPOINT}/1`
+    }).then((response) => {
+      expect(response).to.have.property('status', 200);
+      expect(response.body).to.have.property('id', 1);
+      expect(response.body).to.be.an('object');
+      expect(response.body).to.have.all.keys('id', 'title', 'body', 'userId');
```
Comment (sttaran) [inline cypress/e2e/api/apiTests.spec.js:16]: I would do this assertion first

File: cypress/e2e/api/apiTests.spec.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      expect(response).to.have.property('status', 200);
+      expect(response.body).to.not.be.empty;
+      expect(response.body).to.be.an('array');
+      response.body.forEach(post => {
+        expect(post).to.have.all.keys('id', 'title', 'body', 'userId');
+      });
+    });
+  });
+
+  it('Get posts list from non-existing URL', () => {
```
Comment (sttaran) [inline cypress/e2e/api/apiTests.spec.js:44]: It is rather invalid id then non-existing URL :)

**Review by sttaran — APPROVED** [review]

File: cypress/e2e/api/apiTests.spec.js (hunk 1 above)
Comment (ikoshel) [reply cypress/e2e/api/apiTests.spec.js:1]: yes, I agree with you, in a real project this is exactly what I would do

**Pull request merged**